### DISCUSSION TITLE
Is factory reset broken?

### DISCUSSION THREAD

**Discussion by rfwilliams** [emporia-vue-local/esphome#259] [Q&A] [answered]
I am in the process of editing the yaml file and added the factory reset button using these instructions:

{
Assuming this is NOT dangerous, can I just replace

button:
  - platform: template
    name: "Two Beeps"
    on_press:
      - rtttl.play: "two short:d=4,o=5,b=100:16e6,16e6"
with

button:
  - platform: template
    name: "Two Beeps"
    on_press:
      - rtttl.play: "two short:d=4,o=5,b=100:16e6,16e6"

  - platform: factory_reset
    name: Restart with Factory Default Settings
}

Yet when I try to compile and program I get these error messages:

Compiling .pioenvs/emporia-local/src/main.o
Linking .pioenvs/emporia-local/firmware.elf
/data/cache/platformio/packages/toolchain-xtensa-esp32/bin/../lib/gcc/xtensa-esp32-elf/8.4.0/../../../../xtensa-esp32-elf/bin/ld: .pioenvs/emporia-local/src/main.o:(.literal._Z5setupv+0x148): undefined reference to `vtable for esphome::factory_reset::FactoryResetButton'
/data/cache/platformio/packages/toolchain-xtensa-esp32/bin/../lib/gcc/xtensa-esp32-elf/8.4.0/../../../../xtensa-esp32-elf/bin/ld: .pioenvs/emporia-local/src/main.o:(.literal._Z5setupv+0x14c): undefined reference to `vtable for esphome::factory_reset::FactoryResetButton'
collect2: error: ld returned 1 exit status
*** [.pioenvs/emporia-local/firmware.elf] Error 1
========================= [FAILED] Took 15.42 seconds =========================

Any ideas?

**Comment by blalor** [comment]
Try `esphome clean path/to/config.yaml` and then recompile/re-run. That works for me when I run into that kind of error.

**Comment by rfwilliams** [comment, marked as answer]
Evidently I do not have that command, and I installed samba so I could check the folder and it looks pretty clean to me.

![tmp](https://github.com/emporia-vue-local/esphome/assets/44302132/9b574c14-07d4-41ea-a4a6-bbf186b24cea)

GOT IT!
I had to 'Clean Build Files'.
